revocation/crl/cache/file.go

```diff
@@ -0,0 +1,133 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+const (
+	// tempFileName is the prefix of the temporary file
+	tempFileName = "notation-*"
+)
+
+// fileCache stores in a tarball format, which contains two files: base.crl and
+// metadata.json. The base.crl file contains the base CRL in DER format, and the
+// metadata.json file contains the metadata of the CRL. The cache builds on top
+// of UNIX file system to leverage the file system concurrency control and
+// atomicity.
+//
+// NOTE: For Windows, the atomicity is not guaranteed. Please avoid using this
+// cache on Windows when the concurrent write is required.
+//
+// fileCache doesn't handle cache cleaning but provides the Delete and Clear
+// methods to remove the CRLs from the file system.
+type fileCache struct {
+	dir    string
+	maxAge time.Duration
+}
+
+type FileCacheOptions struct {
+	Dir    string
+	MaxAge time.Duration
+}
+
+// NewFileCache creates a new file system store
+//
+//   - dir is the directory to store the CRLs.
+//   - maxAge is the maximum age of the CRLs cache. If the CRL is older than
+//     maxAge, it will be considered as expired.
+func NewFileCache(opts *FileCacheOptions) (Cache, error) {
```

Comment:
Consider
```suggestion
type FileCache struct {
	// MaxAge <write doc here>
	MaxAge time.Duration

	root    string
}

// NewFileCache creates a new file system store
//
//   - dir is the directory to store the CRLs.
//   - maxAge is the maximum age of the CRLs cache. If the CRL is older than
//     maxAge, it will be considered as expired.
func NewFileCache(root string) (*FileCache, error) {
```

Reply:
In the test, you can ensure whether `*FileCache` implements the `Cache` interface.

Reply:
Same comment applies to MemoryCache.